### DISCUSSION TITLE
如何将TreeView的项自由的排布在Panel（Canvas）上

### DISCUSSION THREAD

**Discussion by TruePluto** [BYJRK/DotNet-Discussions#261] [C# 技术交流]
在WPF中
```
public class Base
{
    public string Name { get; set; } = string.Empty;
    public double Angle { get; set; };
    public double X { get; set; }
    public double Y { get; set; }
}
public class A : Base
{
    public ObservableCollection<B> Bs { get; set; } = new();
}

public class B : Base
{
    public ObservableCollection<C> Cs { get; set; } = new();
}

public class C : Base
{
    public string Des { get; set; } = string.Empty;
}
```
有这样一组类 可以组成一个树形结构，可以通过HierarchicalDataTemplate和DataTemplate显示在TreeView中，
但是我想让他们显示在一个Panel上（ 如Canvas ）位置和角度由X，Y和Angle决定。逻辑上依然是树形结构，但是在Panel上的位置是自由的，由数据自己决定，可重叠。
试着使用过ItemsControl 但是不支持嵌套（子项）。也使用过TreeView但是子项的位置角度无法绑定。
有哪位大神知道如何设置吗？ （可以创建新的Control和Panel）
感谢您的关注

**Comment by TRARS** [comment]
奇怪需求大抵上需[手搓自定义控件](https://mega.nz/file/U2pQnILa#75flh4id4XQSw5tNwt76TwgmokdVyxuH14oDUacW5Ts)来实现
